packages/generator/src/generate.ts

```diff
@@ -2,17 +2,19 @@ import * as pdfLib from '@pdfme/pdf-lib';
 import type { GenerateProps } from '@pdfme/common';
 import { checkGenerateProps, getDynamicTemplate } from '@pdfme/common';
 import { modifyTemplateForTable, getDynamicHeightForTable } from '@pdfme/schemas';
-import { insertPage, preprocessing, postProcessing, getEmbedPdfPages } from './helper.js';
+import { insertPage, preprocessing, postProcessing, getEmbedPdfPages, validateRequiredFields } from './helper.js';
 
 const generate = async (props: GenerateProps) => {
   checkGenerateProps(props);
   const { inputs, template, options = {}, plugins: userPlugins = {} } = props;
   const basePdf = template.basePdf;
 
   if (inputs.length === 0) {
-    throw new Error('inputs should not be empty');
+    throw new Error('inputs should not be empty, pass at least an empty object in the array');
```

Comment:
Could you add the following prefixes that users can distinguish between user-caused errors and system-caused errors?

 `[@pdfme/generator]` .

Reply:
Yes, I will add this.

I wonder if in future we might add a `PdfmeError` class extending `Error` where we can separate the `package` (`generator`, `ui`, `common`, etc) as an attribute. 

This way you could separate the actual error message from the service so that if you did want to expose a message to end users rather than developers you could.

Something to consider for future maybe.

Reply:
100% ggree!